test/test-fft-native.cpp

```diff
@@ -0,0 +1,226 @@
+    AlignedVector64<std::complex<double>> transformed(n);
+    AlignedVector64<std::complex<double>> result(n);
+
+    Inverse_FFT_FromBitReverseRadix2(transformed.data(), operand.data(),
+                                     inv_root_powers.data(), n, &scalar);
+
+    Forward_FFT_ToBitReverseRadix2(result.data(), transformed.data(),
+                                   root_powers.data(), n, &inv_scale);
+
+    for (size_t i = 0; i < n; ++i) {
+      auto tmp = abs(operand[i].real() - result[i].real());
+      ASSERT_TRUE(tmp < 0.5);
+      tmp = abs(operand[i].imag() - result[i].imag());
+      ASSERT_TRUE(tmp < 0.5);
+    }
+  }
+
+  {  // Large Scaled
+    const uint64_t n = 64;
+    const double scale = 1099511627776;  // (1 << 40)
+    const double scalar = scale / static_cast<double>(n);
+    const double inv_scale = static_cast<double>(1.0) / scale;
+    const double data_bound = (1 << 30);
+
+    AlignedVector64<std::complex<double>> operand(n);
+    AlignedVector64<std::complex<double>> transformed(n);
+    AlignedVector64<std::complex<double>> result(n);
+
+    for (size_t i = 0; i < n; i++) {
+      operand[i] = std::complex<double>(
+          GenerateInsecureUniformRealRandomValue(0, data_bound),
+          GenerateInsecureUniformRealRandomValue(0, data_bound));
+    }
+
+    Inverse_FFT_FromBitReverseRadix2(transformed.data(), operand.data(),
+                                     inv_root_powers.data(), n, &scalar);
+
+    Forward_FFT_ToBitReverseRadix2(result.data(), transformed.data(),
+                                   root_powers.data(), n, &inv_scale);
+
+    for (size_t i = 0; i < n; ++i) {
+      double tmp = abs(operand[i].real() - result[i].real());
+      ASSERT_TRUE(tmp < 0.5);
+      tmp = abs(operand[i].imag() - result[i].imag());
+      ASSERT_TRUE(tmp < 0.5);
+    }
+  }
+
+  {  // Very Large Scale
+    const uint64_t n = 64;
+    const double scale = 1.2980742146337069e+33;  // (1 << 110)
+    const double scalar = scale / static_cast<double>(n);
+    const double inv_scale = static_cast<double>(1.0) / scale;
+    const double data_bound = (1 << 20);
+
+    AlignedVector64<std::complex<double>> operand(n);
+    AlignedVector64<std::complex<double>> transformed(n);
+    AlignedVector64<std::complex<double>> result(n);
+
+    for (size_t i = 0; i < n; i++) {
+      operand[i] = std::complex<double>(
+          GenerateInsecureUniformRealRandomValue(0, data_bound),
+          GenerateInsecureUniformRealRandomValue(0, data_bound));
+    }
+
+    AlignedVector64<std::complex<double>> expected = operand;
+
+    Inverse_FFT_FromBitReverseRadix2(transformed.data(), operand.data(),
+                                     inv_root_powers.data(), n, &scalar);
+
+    Forward_FFT_ToBitReverseRadix2(result.data(), transformed.data(),
+                                   root_powers.data(), n, &inv_scale);
+
+    for (size_t i = 0; i < n; ++i) {
+      double tmp = abs(expected[i].real() - result[i].real());
```

Comment:
This is quite often. It may be better to just define your own function based on EXPECT NEAR or the current logic that takes in a complex number and does this comparison.